### PR TITLE
MOSIP-25356

### PR DESCRIPTION
**Pull request by dhanendra06**

### CODE REVIEW

File: id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java [hunk 1]
```diff
@@ -101,12 +101,14 @@ public class CredentialRequestServiceImpl implements CredentialRequestService {
...
+		String requestId=null;
 		List<ServiceError> errorList = new ArrayList<>();
 		ResponseWrapper<CredentialIssueResponse> credentialIssueResponseWrapper = new ResponseWrapper<CredentialIssueResponse>();
 
 		CredentialIssueResponse credentialIssueResponse = null;
-		String requestId = utilities.generateId();
-
+		if(credentialIssueRequestDto.getAdditionalData()!=null) {
+			 requestId= credentialIssueRequestDto.getAdditionalData().containsKey("credential_id") ? credentialIssueRequestDto.getAdditionalData().get("credential_id").toString() : utilities.generateId();
+		}
```
Comment (vishwa-vyom) [inline id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java:111]: This has to be handled in the new endpoint with V2 in the URL where the credentialId will be taken in the request directly

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java (hunk 1 above)
Comment (dhanendra06) [reply id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java]: taken care

File: id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java [hunk 2]
```diff
@@ -156,6 +155,62 @@ public ResponseWrapper<CredentialIssueResponse> createCredentialIssuance(Credent
...
+		List<ServiceError> errorList = new ArrayList<>();
+		ResponseWrapper<CredentialIssueResponse> credentialIssueResponseWrapper = new ResponseWrapper<CredentialIssueResponse>();
+
+		CredentialIssueResponse credentialIssueResponse = null;
+		try{
+			CredentialEntity credential=new CredentialEntity();
+			credential.setRequestId(rid);
+			credential.setRequest(mapper.writeValueAsString(credentialIssueRequestDto));
+			credential.setStatusCode(CredentialStatusCode.NEW.name());
+			credential.setCreateDateTime(LocalDateTime.now(ZoneId.of("UTC")));
```
Comment (LoganathanSekar7627) [inline id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java:171]: You can use `io.mosip.kernel.core.util.DateUtils.getUTCCurrentDateTime()`.

File: id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java [hunk 3]
```diff
@@ -156,6 +155,62 @@ public ResponseWrapper<CredentialIssueResponse> createCredentialIssuance(Credent
...
+			auditHelper.auditError(AuditModules.ID_REPO_CREDENTIAL_REQUEST_GENERATOR,
+					AuditEvents.CREATING_CREDENTIAL_REQUEST, rid, IdType.ID, e);
+			ServiceError error = new ServiceError();
+			error.setErrorCode(CredentialRequestErrorCodes.UNKNOWN_EXCEPTION.getErrorCode());
+			error.setMessage(CredentialRequestErrorCodes.UNKNOWN_EXCEPTION.getErrorMessage());
+			errorList.add(error);
+			LOGGER.error(IdRepoSecurityManager.getUser(), CREDENTIAL_SERVICE, CREATE_CREDENTIAL, ExceptionUtils.getStackTrace(e));
+		} finally {
+			credentialIssueResponseWrapper.setId(EnvUtil.getCredReqServiceId());
+			DateTimeFormatter format = DateTimeFormatter.ofPattern(EnvUtil.getDateTimePattern());
```
Comment (LoganathanSekar7627) [inline id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java:198]: You can use io.mosip.kernel.core.util.DateUtils.getUTCCurrentDateTime(), no need for getting a formatter and format the date time again back to LocalDateTime.

**Review by LoganathanSekar7627 — CHANGES_REQUESTED** [review]

File: id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java (hunk 3 above)
Comment (dhanendra06) [reply id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java]: taken care

File: id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java (hunk 2 above)
Comment (dhanendra06) [reply id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java]: taken care

**Review by vishwa-vyom — APPROVED** [review]
Approving based on Loga's review

**Pull request merged**